src/lib/puzzle/game.js

```diff
@@ -15,6 +15,7 @@ import { createViewBox } from './viewbox';
  * @property {String} color
  * @property {Boolean} locked
  * @property {EdgeMark[]} edgeMarks
+ * @property {EdgeMark[]} reflectEdgeMarks
```

Comment:
I am new to Svelte and thought that I needed to drive the marks from state. But now that I think about it, I can generate these on the fly by looking at state outside the current tile.

Reply:
Since the the edge marks are pulling part of the state from neighbours, I force the neighbour to redraw in `game.toggleEdgeMark`:

```js
		if (self.grid.EDGEMARK_REFLECTS) {
			// force redraw of the neighbour because drawing of connection edgemarks
			// is driven partly by neighbours' state
			const neighbourState = self.tileStates[neighbour];
			neighbourState.set(neighbourState.data);
		}
```

Otherwise half-marks get left behind. I don't know if there is a better way to do this.